tests/spark_autologging/test_spark_disable_autologging.py

```diff
@@ -0,0 +1,115 @@
+import time
+
+import pytest
+
+import mlflow
+import mlflow.spark
+
+from tests.tracking.test_rest_tracking import BACKEND_URIS
+from tests.tracking.test_rest_tracking import tracking_server_uri  # pylint: disable=unused-import
+from tests.tracking.test_rest_tracking import mlflow_client  # pylint: disable=unused-import
+from tests.spark_autologging.utils import (
+    _assert_spark_data_logged,
+    _assert_spark_data_not_logged,
+)
+from tests.spark_autologging.utils import spark_session  # pylint: disable=unused-import
+from tests.spark_autologging.utils import format_to_file_path  # pylint: disable=unused-import
+from tests.spark_autologging.utils import data_format  # pylint: disable=unused-import
+from tests.spark_autologging.utils import file_path  # pylint: disable=unused-import
+
+
+# Note that the following tests run one-after-the-other and operate on the SAME spark_session
```

Comment:
@mohamad-arabi This is awesome! Can we also test the case where there isn't a preexisting spark session and we call `autolog()` with disable=True/False before creating a session?

Reply:
In my opinion that is covered in these 2  test cases: 
https://github.com/dbczumar/mlflow/pull/20/files#diff-7e304d4fffb199878e90000a4dd020fb4d8b5b9a046289a55babd3e44314e643R113

and 

https://github.com/mlflow/mlflow/blob/master/tests/spark_autologging/test_spark_datasource_autologging.py#L228

Reply:
These just test that no exceptions are thrown. They don’t verify that tags are set or not set depending on the disable flag when a session is created after autolog is called. Let me know if you think we already capture that elsewhere